src/manifolds/Grassmann.jl

```diff
@@ -278,12 +278,12 @@ In this formula the $\operatorname{atan}$ is meant elementwise.
 """
 log(::Grassmann, ::Any...)
 
-function log!(M::Grassmann, X, p, q)
+function log!(::Grassmann{n,k}, X, p, q) where {n,k}
     z = q' * p
     At = q' - z * p'
     Bt = z \ At
     d = svd(Bt')
-    return copyto!(X, d.U * Diagonal(atan.(d.S)) * d.Vt)
+    return X .= view(d.U, :, 1:k) * Diagonal(atan.(view(d.S, 1:k))) * view(d.Vt, 1:k, :)
```

Comment:
One other simplification I had noticed in the above lines. They could be changed to
```julia
z = p' * q
A = q - p * z
B = A / z
d = svd(B)
```
And perhaps, depending on what we know about `z` could even do `B = q / z - p`, avoiding allocating `A` entirely.

Reply:
What would we need to know about z for the last line? It looks really sweet!

Reply:
So neither of your proposed code (the for lines or the three lines with avoiding `A`) do pass the tests for the complex case one obtains non-tangent results

````
  Expression: isapprox(M, pts[1], exp(M, pts[2], X2); atol = atolp1p2, rtol = rtolp1p2)
   Evaluated: isapprox(Grassmann(3, 2, ℂ), Complex{Float64}[0.5 + 0.5im 0.5 + 0.5im; 0.5 + 0.5im -0.5 - 0.5im; 0.0 + 0.0im 0.0 + 0.0im], Complex{Float64}[-0.5 - 0.49999999999999994im 0.5000000000000001 + 0.5000000000000001im; -0.5000000000000001 - 0.5im -0.5 - 0.5im; 6.162975822039155e-33 + 3.0814879110195774e-33im 5.551115123125784e-17 + 0.0im]; atol = 2.220446049250313e-13, rtol = 0)
````

It seems to be at least numerically less precise?

Reply:
To be precise, above error message is the one with `A`, the one without fails even a lot, with a check for the tangent vector yields nearly pi instead of zero.

Reply:
`z` is square, so all you'd need is for `z/z == one(z)`. According to the docs `w = z/z` is defined as `w * z = z`, which _could_ produce a pseudoinverse `w != one(z)` if `z` is not invertible. So I think you'd need to know that `z` is invertible.

Reply:
> It seems to be at least numerically less precise?

Hmm, strange. I don't see why that would be. Did you catch the change to the definition of `z`?

Reply:
Oh. Sorry. Maybe I should take a break soon.

Reply:
So to be precise again.
For the new variant with `A` I get the error message from above, which just means its numerically not as stable as the previous variant.
The variant without `A` does the job actually! So not only less allocations but as precise as before. Nice, thanks!

Reply:
Weird, and a little bit confusing. I'll take it though!

Reply:
Well _with_ your one-line-B and without `A` is the variant we now have and it seems as precise as the original, so that's nice.

Reply:
Interestingly, these lines are now just computing the `inverse_retraction!` with `PolarInverseRetraction`.